functional-tests/src/test/groovy/fixtures/FixtureManager.groovy

```diff
@@ -0,0 +1,195 @@
+
+  // Default connection values
+  private String host = 'localhost';
+  private int port = 27017;
+  private String databaseName = 'mem-dev-func';
+
+  // Stores fixture data that is inserted via this manager.
+  private ArrayList fixtureDataInserted = [];
+
+  /**
+   * A data fixture manager.
+   *
+   * This manager handles connecting to the mongo database and contains helper methods for inserting and
+   * removing data fixtures.
+   *
+   * When finished with this manager, it is recommneded that you call FixtureManager#removeAll() which will
+   * remove any fixture data that was added by this instance of the manager.
+   */
+  FixtureManager(final String host = null, final Integer port = null, final String databaseName = null) {
+    def env = System.getenv()
+    if(host) {
+      this.host = host;
+    } else if (env['MONGODB_FUNC_HOST']) {
+      this.host = env['MONGODB_FUNC_HOST'];
+    }
+
+    if (port) {
+      this.port = port;
+    } else if (env['MONGODB_FUNC_PORT']) {
+      this.port = env['MONGODB_FUNC_PORT'] as int;
+    }
+
+    if (databaseName) {
+      this.databaseName = databaseName;
+    } else if (env['MONGODB_FUNC_DATABASE'] ) {
+      this.databaseName = env['MONGODB_FUNC_DATABASE'];
+    }
+
+    this.client = new MongoClient(this.host, this.port);
+
+    this.db = client.getDB(this.databaseName);
+  }
+
+  /**
+   * @return the database instance used by this manager.
+   */
+  public getDB() {
+    return this.db;
+  }
+
+  /**
+   * Gets the fixture and returns the json object.
+   * @param fixture file name (required).
+   * @return an ArrayListMap of the fixture data or null if no data file is found.
+   */
+  public Object getFixture(String fixtureName) {
+    if(!fixtureName.endsWith(".json")) {
+      fixtureName += ".json";
+    }
+
+    try {
+      def String fixtureFilePath = [fixtureDataFolderPath, fixtureName].join('/');
+      def String fixtureFileJson = new File(fixtureFilePath).getText()
+      return new JsonSlurper().parseText(fixtureFileJson);
+    } catch (FileNotFoundException e) {
+      return null;
+    }
+  }
+
+  /**
+   * Inserts all fixture data contained within the matching data file, if one exists.
+   * @param fixtureName the name of the json data file.  Specifying the ".json" file extension is optional.
+   */
+  public insertFixture(final String fixtureName) {
+    Object fixtureData = getFixture(fixtureName);
+    insert(fixtureData);
+  }
+
+  /**
+   * Inserts all fixture data contained within the passed fixtureData parameter.
+   * @param fixtureData a fixture data object.
+   *
+   * Example:
+   *  fixtureData = [
+   *    "collection": "someCollection",
+        "records": [
+          [
+            "firstName": "jon",
+            "lastName": "smith",
+            "address": [
+              "street": "123 fake street",
+              "postal": "A1A 1A1"
+            ]
+          ]
+        ]
+   *  ]
+   */
+  public insertFixture(final Object fixtureData) {
```

Comment:
Does the overloading work correctly? Just curious a `String` is, technically an `Object`...